### DISCUSSION TITLE
Assign IP to an Interface - Netbox

### DISCUSSION THREAD

**Discussion by ISAYLG** [netbox-community/netbox#7870] [Q&A] [answered]
Hi,
I'm trying to cretae new IPs and assign them to the relevent inetrfaces. as part of my python code, I retrieve the device ID and inetrface ID and then create the IP putting also the device and interface attributes. but when I get back to netbox, under my interface, it doesnt have any IP assigned to it, so the new IP gets created but it's not associated to any inetrface. Is this doable as currently not working for me?

new_ip = nb.ipam.ip_addresses.create(
    address = '10.10.6.1/32',
    status = 'active',
    role = 'loopback',
    interface = id,
)

**Comment by candlerb** [comment, marked as answer]
This is a generic relation, because an IP address can be assigned to either an Interface (on a Device) or a VMInterface.  You need to set `assigned_object_type` and `assigned_object_id`

Here is some code which creates a virtual machine + interface + primary IPv4 + primary IPv6

```
#!/usr/bin/python3
import pynetbox
import sys

name=sys.argv[1]
domain=sys.argv[2]
ipv4=sys.argv[3]
ipv6=sys.argv[4]
CLUSTER = "mycluster"
domain_name = None
if domain:
    domain_name="%s.%s" % (name,domain)

nb = pynetbox.api('https://netbox.example.com',
        token='XXXXXXXX',
    )
vm = nb.virtualization.virtual_machines.create(dict(
        name=name,
        cluster={"name": CLUSTER},
    ))
intf = nb.virtualization.interfaces.create(dict(
    name="eth0",
    virtual_machine={"id":vm.id},
    ))  # TODO: set MAC address?
addrs = nb.ipam.ip_addresses.create([
    dict(address=ipv4, family=4, assigned_object_type="virtualization.vminterface", assigned_object_id=intf.id, dns_name=domain_name),
    dict(address=ipv6, family=6, assigned_object_type="virtualization.vminterface", assigned_object_id=intf.id, dns_name=domain_name),
])
vm.primary_ip4 = dict(id=addrs[0].id)
vm.primary_ip6 = dict(id=addrs[1].id)
vm.save()
```

Note the addresses created with `assigned_object_type` (which for you would be `dcim.interface`) and `assigned_object_id`.
